ruby.md

```diff
@@ -6,40 +6,65 @@ Answer the following questions. First, without external resources. Challenge you
 1. In what ways are JavaScript and Ruby similar? In what ways are they different?
 
   Your answer:
+They both have a function, loop, argument statement, blocks.
+Javascript uses console.log whereas Ruby use puts. Ruby end the methods with "end" whereas JS end with return, but Ruby can use return, but must end the blocks.  
+Ruby uses def while JS uses variable and const. 
 
   Researched answer:
+  JavaScript can be used as front-end and back-end language using the same language whereas, Ruby is used as back-end programming language.JavaScript is more scalable than Ruby as it is 20 times faster than Ruby in some cases. JavaScript has no ability to monitor memory utilization whereas Ruby has the ability.
+
+
 
 
 
 2. What is a hash?
 
   Your answer:
+  hash is a similar concept of the relationship between two entities such as animals => 4 legs (animals with only 4 legs).
 
   Researched answer:
+  A Hash is a collection of key-value pairs like this: "employee" = > "salary". It is similar to an Array, except that indexing is done via arbitrary keys of any object type, not an integer index.
+The order in which you traverse a hash by either key or value may seem arbitrary and will generally not be in the insertion order. If you attempt to access a hash with a key that does not exist, the method will return nil.
+
+
+
 
 
 
 3. What is the testing framework used in Ruby? Describe the process of setting up the testing environment.
 
   Your answer:
+  Testing framework in ruby is a testing using to test your work enviornment in ruby.
 
   Researched answer:
+test-unit (Test::Unit) is unit testing framework for Ruby, based on xUnit principles. It allows writing tests, checking results and automated testing in Ruby.
+
+
 
 
 
 4. Name three possible relationships between objects?
 
   Your answer:
+  Class
+  Blocks
+  Lambdas
 
   Researched answer:
+Blocks, lambdas, Class
+
```

Comment:
Relationships would be has_a, has_many, is_a, belongs_to  -  just like the whiteboarding we did yesterday.